src/utils/__tests__/style-formatter.spec.js

```diff
@@ -31,133 +34,145 @@ describe("style-formatter", () => {
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes(`color: ${someColorExpression}`)).toBe(true);
     });
 
     it("should return font-weight: bold when bold is selected", () => {
       style.font.style = {
         bold: true,
       };
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes("font-weight: bold")).toBe(true);
     });
 
     it("should return font-style: italic when italic is selected", () => {
       style.font.style = {
         italic: true,
       };
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes("font-style: italic")).toBe(true);
     });
     // background
     it("should return specified background color", () => {
       style.background.color = someColor;
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes(`background-color: ${someColor}`)).toBe(true);
     });
 
     it("should return specified background color from expression", () => {
       style.background.colorExpression = someColorExpression;
       style.background.useColorExpression = true;
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes(`background-color: ${someColorExpression}`)).toBe(true);
     });
 
     it("should return default background color when color is none", () => {
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes("background-color: myPrimaryColor")).toBe(true);
     });
 
-    it("should return specified image url when the background mode is set on media", () => {
+    it("should set the image url when the image url exists", async () => {
       style.background.useImage = false;
       style.background.mode = "media";
       style.background.url.qStaticContentUrl = { qUrl: someUrl };
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
-      expect(formattedStyle.includes(`background-image: url('https://localhost${someUrl}')`)).toBe(true);
+      imageUrl = await new Promise((resolve) => {
+        setTimeout(() => resolve(`https://localhost${someUrl}`), 1000);
+      });
+      await waitFor(() => expect(imageUrl).toBe(`https://localhost${someUrl}`));
```

Comment:
This is just verifying the promise that was created on the previous line.

What is that you really want to test here?

Reply:
Just assign the url directly instead of waiting for it to be done `imageUrl = `https://localhost${someUrl}`